### PR TITLE
Add description arg

### PR DESCRIPTION
**Pull request by enerrio**
# Description

This PR adds an optional description arg to the `create_dataset` and `create_dataset_from_buffers` functions which allow a user to include a brief description of the dataset they are creating. If a description is not provided then a warning will be displayed to the user stating it is recommended to include one.

A minimal working example displaying usage of description:
```python
import gymnasium as gym
from minari.data_collector import DataCollector

env = gym.make("Blackjack-v1")
env = DataCollector(env)

for episode in range(10):
    done = False
    _, info = env.reset()  # {}
    while not done:
        action = env.action_space.sample()
        _, _, terminated, truncated, _ = env.step(action)
        done = terminated or truncated

dataset = env.create_dataset(
    dataset_id="blackjack-random-v1",
    algorithm_name="Random-Policy",
    code_permalink="dummy-permalink",
    author="Farama",
    author_email="dummy-email",
    minari_version="==0.4.3",
    description="A random policy for 10 fixed episodes" # commenting out this line will result in a warning to user
)
```

Fixes # (issue), Depends on # (pull request)

## Type of change

> Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

### Screenshots

N/A

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have run `pytest -v` and no errors are present.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I solved any possible warnings that `pytest -v` has generated that are related to my code to the best of my knowledge.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: minari/utils.py [hunk 1]
```diff
@@ -385,6 +386,15 @@ def _generate_dataset_metadata(
     else:
         dataset_metadata["algorithm_name"] = algorithm_name
 
+    if description is None:
+        print("here")
```
Comment (younik) [inline minari/utils.py:390]: Can you remove this print?

File: minari/data_collector/data_collector.py [hunk 2]
```diff
@@ -341,6 +342,7 @@ def create_dataset(
                                                                             `ref_max_score` and `expert_policy` can't be passed at the same time. Default to None
             num_episodes_average_score (int): number of episodes to average over the returns to compute `ref_min_score` and `ref_max_score`. Default to 100.
             minari_version (Optional[str], optional): Minari version specifier compatible with the dataset. If None (default) use the installed Minari version.
+            description (Optional[str], optional): description of the dataset being created. Defaults to None.
```
Comment (younik) [inline minari/data_collector/data_collector.py:345]: Actually, we should use `(Optional[str]l)`, also the line above is wrong; can you fix it?

File: minari/utils.py [hunk 3]
```diff
@@ -506,6 +517,7 @@ def create_dataset_from_buffers(
         action_space (Optional[gym.spaces.Space]): action space of the environment. If None (default) use the environment action space.
         observation_space (Optional[gym.spaces.Space]): observation space of the environment. If None (default) use the environment observation space.
         minari_version (Optional[str], optional): Minari version specifier compatible with the dataset. If None (default) use the installed Minari version.
+        description (Optional[str], optional): description of the dataset being created. Defaults to None.
```
Comment (younik) [inline minari/utils.py:520]: same as other comment

**Review by younik — CHANGES_REQUESTED** [review]
Thank you @enerrio!
I just added a couple of comments

File: minari/utils.py (hunk 1 above)
Comment (enerrio) [reply minari/utils.py]: oop. fixed!

File: minari/data_collector/data_collector.py (hunk 2 above)
Comment (enerrio) [reply minari/data_collector/data_collector.py:345]: Do you mean wanting the description to be required instead of defaulting to None?

Comment (younik) [reply minari/data_collector/data_collector.py:345]: No, it is just for the documentation `Optional[str]l` means it is optional already, so we can remove `, optional`

Comment (enerrio) [reply minari/data_collector/data_collector.py:345]: I see throughout the repo many of the other optional args also have the same format of `(Optional[<dtype>], optional)` in docstrings. Should these be kept for consistency?

Comment (younik) [reply minari/data_collector/data_collector.py:345]: Yes, actually, we should standardize it across all the documentation, so let's do it in another PR.

**Review by younik — APPROVED** [review]

**Pull request merged**